src/main/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMyBatisRepository.java

```diff
@@ -107,11 +109,35 @@ public Map<String, ClinicalDataType> getClinicalAttributeDatatypeMap() {
         
         return attributeDatatypeMap;
     }
+    
+    public static List<CaseListDataCount> mergeCaseListCounts(List<CaseListDataCount> counts) {
```

Comment:
@haynescd i'm assuming that this should go in some util?  let me know the right place or if this is ok to just have it hanging out here.
FYI, i don't love that this merging is done at repository level because it suggest that the result of the underlying mapper is legitimate.  it's not.  the result must be merged. but i can't figure out a way to do that operation in SQL.  ideally, it would be done in the mapper, but the mapper methods have no java implementation so ... what to do?

Reply:
So, I would suggest doing this at the service layer.

Creating a class CaseListService? Or whatever where this logic lived.


And the Repository would return getCaseListDataCountsPerStudy()

That way the mapper is still correct and you can do the business logic of merging in the service layer